### PR TITLE
Reviewed and updated the read and write examples documentation.

### PR DESCRIPTION
**Pull request by ksenteer**
Provided stylistic and grammar improvements based on the technical documentation standards and best practices. Edited misprints.

Relates to: OLPEDGE-603

Signed-off-by: Kateryna Osadchenko <ext-kateryna.osadchenko@here.com>

### CODE REVIEW

File: docs/dataservice-read-catalog-example.md [hunk 1]
```diff
@@ -1,8 +1,10 @@
 # Read example
 
-This example shows how to retrieve catalog metadata, partition metadata, and partition data using the HERE OLP Edge SDK C++.
+This example shows how to retrieve catalog metadata, partition metadata, and
+partition data using the HERE OLP Edge SDK C++.
 
-Before you run the example, you have to replace the placeholders in `examples/dataservice-read/example.cpp` with your app key and app secret:
+Before you run the example, log in by using your platform credentials, which you
```
Comment (leonid-ziskel) [inline docs/dataservice-read-catalog-example.md]: Why do you need to login?

Comment (ksenteer) [reply docs/dataservice-read-catalog-example.md]: You need to authorize by replacing the placeholders in `examples/dataservice-read/example.cpp` with your app access key id, secret access key, catalog HRN and layer. Let me adjust this description and remove "log in by using your platform credentials" from READ and WRITE examples.

File: docs/dataservice-read-catalog-example.md [hunk 2]
```diff
@@ -30,27 +33,37 @@ To execute the example, run:
...
 
 ### Build HERE OLP Edge SDK C++
 
-First, before building, you need to configure the SDK with `EDGE_SDK_BUILD_EXAMPLES` set to `ON`, the path to Android NDK's toolchain file set via the `CMAKE_TOOLCHAIN_FILE` variable, and, optionally, the [NDK-specific CMake variables](https://developer.android.com/ndk/guides/cmake#variables).
+Before you build the example, complete the following:
+* Configure the SDK with `EDGE_SDK_BUILD_EXAMPLES` set to `ON`.
+* Disable tests with `EDGE_SDK_ENABLE_TESTING` set to `OFF`.
+* Specify the path to Android NDK's toolchain file set via the `CMAKE_TOOLCHAIN_FILE`
+variable.
+* In case you want to build Edge SDK for a specific Android platform, use `-DANDROID_PLATFORM CMake` flag, and `-DANDROID_ABI`, if you want to build for specific Android
```
Comment (leonid-ziskel) [inline docs/dataservice-read-catalog-example.md]: the last two points belong together. I don't think it makes sense to separate them

File: docs/dataservice-read-catalog-example.md [hunk 3]
```diff
@@ -69,55 +83,74 @@ cd examples/dataservice-read/android
 
 Alternatively, you can use the Android Studio IDE by opening the `build/examples/dataservice-read/android/build.gradle` script.
 
-After installing and running the `dataservice_read_example` apk, you should see the `Reading the partition data from the specified catalog completed successfully` message in the main UI screen if you provided correct keyId and secretKey. If you encountered an error, please check the device's logcat for the error message.
```
Comment (leonid-ziskel) [inline docs/dataservice-read-catalog-example.md]: why is this section removed?

File: docs/dataservice-read-catalog-example.md [hunk 4]
```diff
@@ -69,55 +83,74 @@ cd examples/dataservice-read/android
...
+the `<olp-edge-sdk-root>/README.md`.
+* Replace the placeholders in `examples/dataservice-read/example.cpp` with your
+app access key id and secret access key.
 
 ### Build HERE OLP Edge SDK C++
 
-First, before building, you need to configure the HERE OLP Edge SDK C++ with `EDGE_SDK_BUILD_EXAMPLES` set to `ON`, (optionally) disable tests with `EDGE_SDK_ENABLE_TESTING` set to `OFF`, and specify the path to the iOS toolchain file shipped together with the SDK and located under `<olp-edge-sdk-root>/cmake/toolchains/iOS.cmake`:
+Before you build the example, complete the following:
+* Configure the HERE OLP Edge SDK C++ with `EDGE_SDK_BUILD_EXAMPLES` set to `ON`.
+* Disable tests with `EDGE_SDK_ENABLE_TESTING` set to `OFF`.
```
Comment (leonid-ziskel) [inline docs/dataservice-read-catalog-example.md]: This is an optional step as pointed out in the original text

File: docs/dataservice-write-example.md [hunk 5]
```diff
@@ -32,27 +35,31 @@ To execute the example, run:
...
+* Set up the Android environment.
+* Replace the placeholders in `examples/dataservice-read/example.cpp` with your app access key id, secret access key, catalog HRN and layer.
 
 ### Build HERE OLP Edge SDK C++
 
-First, before building, you need to configure the SDK with `EDGE_SDK_BUILD_EXAMPLES` set to `ON`, the path to Android NDK's toolchain file set via the `CMAKE_TOOLCHAIN_FILE` variable, and, optionally, the [NDK-specific CMake variables](https://developer.android.com/ndk/guides/cmake#variables).
+Before you build the example, complete the following:
+* Configure the SDK with `EDGE_SDK_BUILD_EXAMPLES` set to `ON`.
+* Specify the path to Android NDK's toolchain file set via the `CMAKE_TOOLCHAIN_FILE` variable.
+* In case you want to build Edge SDK for a specific Android platform, use `-DANDROID_PLATFORM CMake` flag, and `-DANDROID_ABI`, if you want to build for specific Android architecture. For more details, see [NDK-specific CMake variables](https://developer.android.com/ndk/guides/cmake#variables).
```
Comment (leonid-ziskel) [inline docs/dataservice-write-example.md]: two last points belong to Android. They should be one point

File: docs/dataservice-write-example.md [hunk 6]
```diff
@@ -166,11 +195,13 @@ Then pass it to the `StreamLayerClient` via `PublishData` method:
...
-The execution result is a `CancellableFuture` that contains `PublishDataResponse` object. The `PublishDataResponse` class holds details of the completed operation, and should be used to determine operation success and access resultant data.
+The execution result is a `CancellableFuture` that contains `PublishDataResponse` object.
+The `PublishDataResponse` class holds details of the completed operation and is used
+to determine operation success and access resultant data.
 
 * `IsSuccessful`: Returns true if this response is considered successful, false otherwise.
-* `GetResult`: Returns the resultant data (`olp::dataservice::write::PublishDataResult`) in the case that the operation is successful.
-* `GetError`: Contains information about the error that occurred in the case of error in a `olp::client::ApiError` object.
+* `GetResult`: Returns the resultant data (`olp::dataservice::write::PublishDataResult`) in case of a successful operation.
+* `GetError`: Contains an error information as a result of an error in the `olp::client::ApiError` object.
```
Comment (leonid-ziskel) [inline docs/dataservice-write-example.md]: maybe "the" error information?

File: docs/dataservice-read-catalog-example.md (hunk 3 above)
Comment (leonid-ziskel) [reply docs/dataservice-read-catalog-example.md]: Sounds good

File: docs/dataservice-read-catalog-example.md [hunk 7]
```diff
@@ -1,8 +1,10 @@
...
-This example shows how to retrieve catalog metadata, partition metadata, and partition data using the HERE OLP Edge SDK C++.
+This example shows how to retrieve catalog metadata, partition metadata, and
+partition data using the HERE OLP Edge SDK C++.
 
-Before you run the example, you have to replace the placeholders in `examples/dataservice-read/example.cpp` with your app key and app secret:
+Before you run the example, log in by using your platform credentials, which you
+can find on the platform Apps and keys page. Replace the placeholders in `examples/dataservice-read/example.cpp` with your app access key id and secret access key:
 
 ```cpp
 const std::string gKeyId(""); // your here.access.key.id
```
[the remark this answers is not in the document]
Comment (ksenteer) [reply docs/dataservice-read-catalog-example.md]: Is this change also relevant for gKeySecret > kKeySecret throughout the docs?

File: docs/dataservice-read-catalog-example.md (hunk 4 above)
Comment (ksenteer) [reply docs/dataservice-read-catalog-example.md]: Got it

File: docs/dataservice-write-example.md (hunk 5 above)
Comment (ksenteer) [inline docs/dataservice-write-example.md]: Specific options are described.

**Comment by ksenteer** [comment]
@andescu @leonid-ziskel when I squashed, and force pushed my commits this #99 pull request got closed.
Here's the link to a new pull request will all the updates according to reviewers' comments: https://github.com/heremaps/here-olp-edge-sdk-cpp/pull/107/files

Please review and let me know if we can merge it/ Thanks!

**Pull request closed without merging**